Mlem/Models/Trackers/Post Tracker.swift

```diff
@@ -296,9 +296,16 @@ class PostTracker: ObservableObject {
         
         // perform real save
         do {
-            let response = try await postRepository.savePost(postId: post.postId, shouldSave: shouldSave)
-            await update(with: response)
-            return response
+            let saveResponse = try await postRepository.savePost(postId: post.postId, shouldSave: shouldSave)
+            await update(with: saveResponse)
+
+            if shouldSave {
+              //FIXME: setting `isLoading` forcefully to false doesn't feel very good.
+              isLoading = false
+              return await voteOnPost(post: saveResponse, inputOp: .upvote)
```

Comment:
Rather than calling the existing `voteOnPost`, we should:
1. Perform save state faking alongside save status (can just add a `votes` parameter to the state fake constructor on 293 with the updated vote state)
2. Call `postRepository.ratePost` directly and update with the response

This does a few things:
- Avoids triggering vote haptic
- Avoids the forced `isLoading = false`
- Fixes vote toggling behavior (see main review comment)

Reply:
Thank you @EricBAndrews! Could you please have another look if I understood you correctly? Thanks in advance